lib/app/admin/data-reset/test/remove-allItemsFromList-spec.js

```diff
@@ -0,0 +1,130 @@
+var sinon = require('sinon');
+var async = require('async');
+require('sinon-as-promised');
+
+var remove = require('./../remove');
+var sampleData = require('./../data-demo/index');
+
+
+describe('Test remove', function() {
+  var mediatorRequestStub = sinon.stub();
+  var mediatorMock = {request: mediatorRequestStub};
+
+  beforeEach(function() {
+    mediatorMock.request.reset();
+
+  });
+
+  it('Should remove all from files list', function(done) {
+    async.eachSeries(sampleData.file(), function(file, eachMsgCb) {
```

Comment:
Don't use async anymore, at least when mixed with promises. And especially in this case, as if your removeAllItemsFromList() function failed, then you'd never call one of the callbacks, the test would mysteriously timeout, and the poor dev who broke it will be left scratching their heads for a while.

Bluebird has several utility functions, particularly for dealing with collections, and this is the main reason why @evanshortiss and I advocate for it so much. In this case, you'll want to use [Promise.mapSeries](http://bluebirdjs.com/docs/api/promise.mapseries.html) instead of asyncEachSeries. In each function, you would just `return remove.removeAllItemsFromList(mediatorMock, [file], 'file');` And make sure you get rid of the `done` mocha callback, and just return the value from Promise.mapSeries().